### PR TITLE
Cd32 menu

### PR DESCRIPTION
**Pull request by AndyLoft**
enable the use of joystick/cd32 gamepad in the main menu

### CODE REVIEW

**Comment by AndyLoft** [comment]
feel free to point me in the right direction if you see something wrong/unnecessary or just plain dumb.

File: ab3d2_source/menu/menunb.s [hunk 1]
```diff
@@ -977,7 +993,15 @@ mnu_waitmenu:;out: d0=Selection number
 .sliderl:
 				moveq.l	#42,d1
 				bra.s	.cpcont
-
+.cd32:
+				clr.b	Plr1_Keys_b
+				clr.b	Plr1_Path_b
+				clr.b	Plr1_Mouse_b
+				st		Plr1_Joystick_b
+				clr.b	Plr2_Keys_b
```
Comment (0xABADCAFE) [inline ab3d2_source/menu/menunb.s:1001]: Do we need to manipulate the Plr2 state here?

Comment (AndyLoft) [reply ab3d2_source/menu/menunb.s:1001]: I do not know. I will try and test it in 2 player later.

File: ab3d2_source/c/game_preferences.c [hunk 2]
```diff
@@ -67,6 +67,7 @@ void game_ApplyPreferences(void)
         Vid_BrightnessOffset_w  = Prefs_BrightnessOffset_RTG_w;
         Vid_GammaLevel_b        = Prefs_GammaLevel_RTG_b;
     } else {
+		Vid_DoubleHeight_b      = Prefs_PixelMode_b;
```
Comment (0xABADCAFE) [inline ab3d2_source/c/game_preferences.c:70]: Since this is now in both branches of the if/else, it might as well be extracted out and put before it.

Comment (AndyLoft) [reply ab3d2_source/c/game_preferences.c:70]: I will sort that out if you find no other issues after testing.

Comment (AndyLoft) [reply ab3d2_source/c/game_preferences.c]: done.

**Comment by 0xABADCAFE** [comment]
Looks good to me. I should probably update the documentation around the config file later.

**Pull request merged**